Failed nightly macOS build

@zisoft I guess that's for you.

Tonight's nightly build failed due to a graphicsmagick rebuild problem in the macOS job.
Since the log for this build will eventually expire, I quote the relevant excerpt below:

```
Run export HOMEBREW_NO_INSTALL_FROM_API=1
  export HOMEBREW_NO_INSTALL_FROM_API=1
  
  gm_formula=`brew edit --print-path graphicsmagick`
  cat $gm_formula | sed '/--with-modules/d' > gm_formula.tmp
  mv gm_formula.tmp $gm_formula
  
  brew reinstall --build-from-source graphicsmagick
  shell: /bin/bash -e {0}
  env:
    DEVELOPER_DIR: /Applications/Xcode_13.[2](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:2).1.app/Contents/Developer
    CC: cc
    CXX: c++
    MACOSX_DEPLOYMENT_TARGET: 11.[3](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:3)
    SRC_DIR: /Users/runner/work/darktable/darktable/src
    BUILD_DIR: /Users/runner/work/darktable/darktable/src/build
    INSTALL_PREFIX: /Users/runner/work/darktable/darktable/src/build/macosx
    ECO: -DBINARY_PACKAGE_BUILD=ON -DBUILD_CURVE_TOOLS=ON -DBUILD_NOISE_TOOLS=ON
    CMAKE_BUILD_TYPE: Release
    GENERATOR: Ninja
    TARGET: skiptest
Warning: edit is a developer command, so Homebrew's
developer mode has been automatically turned on.
To turn developer mode off, run:
  brew developer off

Warning: building from source is not supported!
You're on your own. Failures are expected so don't create any issues, please!
==> Fetching graphicsmagick
==> Downloading https://downloads.sourceforge.net/project/graphicsmagick/graphicsmagick/1.3.[4](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:4)0/GraphicsMagick-1.3.40.tar.xz
==> Downloading from https://phoenixnap.dl.sourceforge.net/project/graphicsmagick/graphicsmagick/1.3.40/GraphicsMagick-1.3.40.tar.xz
==> Reinstalling graphicsmagick 
==> ./configure --prefix=/usr/local/Cellar/graphicsmagick/1.3.40_2 --disable-ope
==> make install
Last 1[5](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:5) lines from /Users/runner/Library/Logs/Homebrew/graphicsmagick/02.make:
    ~~~~~~~~~ ^
coders/jpeg.c:14[6](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:6)6:26: error: use of undeclared identifier 'JPROC_PROGRESSIVE'
    jpeg_info.process == JPROC_PROGRESSIVE ? LineInterlace : NoInterlace;
                         ^
coders/jpeg.c:146[7](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:7):32: error: no member named 'process' in 'struct jpeg_decompress_struct'
  image->compression=jpeg_info.process == JPROC_LOSSLESS ?
                     ~~~~~~~~~ ^
coders/jpeg.c:[14](https://github.com/darktable-org/darktable/actions/runs/5639374704/job/15274507014#step:4:14)67:43: error: use of undeclared identifier 'JPROC_LOSSLESS'
  image->compression=jpeg_info.process == JPROC_LOSSLESS ?
                                          ^
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I. -g -O2 -Wall -D_THREAD_SAFE -c coders/jxl.c  -fno-common -DPIC -o coders/.libs/magick_libGraphicsMagick_la-jxl.o
4 errors generated.
make[1]: *** [coders/magick_libGraphicsMagick_la-jpeg.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [install] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

Error: Process completed with exit code 1.
```

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -264,7 +264,7 @@ jobs:
           retention-days: 2
 
   macOS:
-    if: 0 && (github.repository == 'darktable-org/darktable' || github.event_name == 'workflow_dispatch')
+    if: github.repository == 'darktable-org/darktable' || github.event_name == 'workflow_dispatch'
     name: Nightly darktable macOS
     runs-on: ${{ matrix.build.os }}
     strategy:
@@ -305,10 +305,15 @@ jobs:
           brew tap Homebrew/bundle
           cd src/.ci
           export HOMEBREW_NO_INSTALL_UPGRADE=1
+          export HOMEBREW_NO_INSTALL_CLEANUP=1
           brew bundle --verbose || true
           # handle keg-only libs
           brew link --force libomp
           brew link --force libsoup@2
+      - name: Downgrade libjpeg-turbo
+        run: |
+          brew uninstall --ignore-dependencies libjpeg-turbo
+          brew link libjpeg-turbo --overwrite
       - name: Rebuild graphicsmagick without modules
         run: |
           export HOMEBREW_NO_INSTALL_FROM_API=1
@@ -360,8 +365,7 @@ jobs:
       # We need write permission to update the nightly tag
       contents: write
     runs-on: ubuntu-latest
-    #needs: [AppImage, Windows, macOS]
-    needs: [AppImage, Windows]
+    needs: [AppImage, Windows, macOS]
     steps:
       - name: Download AppImage artifact
         uses: actions/download-artifact@v3
@@ -371,10 +375,10 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: Darktable.Nightly.Windows
-      # - name: Download macOS artifact
-      #   uses: actions/download-artifact@v3
-      #   with:
-      #     name: Darktable.Nightly.macOS
+      - name: Download macOS artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: Darktable.Nightly.macOS
       - name: Update nightly release
         uses: andelf/nightly-release@main
         env:
@@ -402,5 +406,5 @@ jobs:
             __Please help us improve Darktable by reporting any issues you encounter!__ :wink:
           files: |
             Darktable-*.AppImage*
-            # darktable-*.dmg
+            darktable-*.dmg
             darktable-*.exe

